sdks/java/io/common/src/test/java/org/apache/beam/sdk/io/common/IOITHelper.java

```diff
@@ -36,4 +42,55 @@ public static String getHashForRecordCount(int recordCount, Map<Integer, String>
     }
     return hash;
   }
+
+  public static void retry(RetryFunction function, int attempts, long delay)
+      throws InterruptedException {
+    ArrayList<Exception> errorList = new ArrayList<Exception>();
+    int countAttempts = 0;
+
+    while (countAttempts < attempts) {
+      try {
+        function.run();
+        return;
+      } catch (Exception e) {
+        LOG.warn(
+            "Attempt #{} of {} threw exception: {}", countAttempts + 1, attempts, e.getMessage());
+        errorList.add(e);
+        countAttempts++;
+        if (countAttempts == attempts) {
+          throw RetryException.composeErrors(errorList);
```

Comment:
I think just throwing the last exception will be cleaner (and log previous once). We can also get rid of "RetryException".

Reply:
Ok, good point! Thanks